Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,309 @@
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Sebastien Loriot, Maxime Gimeno
+
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_EXTRUDE_H
+#define CGAL_POLYGON_MESH_PROCESSING_EXTRUDE_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+
+#include <CGAL/Polygon_mesh_processing/orientation.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/boost/graph/named_function_params.h>
+#include <CGAL/boost/graph/copy_face_graph.h>
+#include <CGAL/Kernel_traits.h>
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <vector>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+namespace extrude_impl{
+
+template<typename PMAP, typename Vector>
+struct Const_dist_translation{
+  Const_dist_translation(PMAP map, const Vector& dir, const double d)
+    :map(map), dir(dir), d(d){}
+  
+  template<typename VertexDescriptor, typename U>
+  void operator()(const VertexDescriptor vd, const U&)
+  {
+    typename boost::property_traits<PMAP>::value_type p = get(map, vd) + d*dir;
+    put(map, vd, p);
+  }
+  
+  PMAP map;
+  Vector dir;
+  double d;
+};
+
+struct Identity_functor
+{
+  template<typename T, typename U>
+  void operator()(const T&, const U&){}
+};
+}//end extrude_impl
+
+/**
+ * \ingroup PMP_meshing_grp
+ * Extrudes the open mesh input and put the result in output. The mesh generated is a closed mesh with 
+ * a bottom and top part, both having the same graph combinatoric as input(except that the orientation 
+ * of the faces of the bottom part are reversed). The bottom and the top parts are connected by a 
+ * triangle strip between each boundary cycles. The coordinates of the points associated to the 
+ * vertices of the bottom and top part are first initialized to the same value as the corresponding 
+ * vertices of input. Then for each vertex, a call to bot and top is done for the vertices of the 
+ * bottom part and the top part, respectively.
+ * @tparam InputMesh a model of the concept `FaceListGraph`
+ * @tparam OutputMesh a model of the concept `MutableFaceListGraph`
+ * @tparam NamedParameters1 a sequence of \ref pmp_namedparameters "Named Parameters" for `InputMesh`
+ * @tparam NamedParameters2 a sequence of \ref pmp_namedparameters "Named Parameters" for `OutputMesh`
+ * @tparam BottomFunctor a functor with a function 
+ * `void operator()(boost::graph_traits<InputMesh>::vertex_descriptor,
+ * boost::graph_traits<OutputMesh>::vertex_descriptor);
+ * @tparam TopFunctor a functor with a similar operator() as `BottomFunctor`.
+ * @param input the open triangulated `InputMesh` to extrude.
+ * @param output the `OutputMesh` containing the result of the extrusion.
+ * @param bot a `BottomFunctor` that will apply a transformation to all points of 
+ * `input` in order to create the first offsetted part of the extrusion. 
```

Comment:
Look at the information provided to the user. Is it sufficient for a user to write its functor?

Reply:
The user knows the signature of the operator() and what it is supposed to do. What more does (s)he need ?

Reply:
maybe what the parameters corresponds to? If I'm using a descriptor from mesh A with mesh B I'm pretty sure the user will be in trouble.